Libraries/Opc.Ua.Security.Certificates/Extensions/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,340 @@
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension(string oid, byte[] rawData, bool critical)
+        :
+            this(new Oid(oid, kFriendlyName), rawData, critical)
+        {
+        }
+
+        /// <summary>
+        /// Build the X509 Authority Key extension.
+        /// </summary>
+        /// <param name="subjectKeyIdentifier">The subject key identifier</param>
+        public X509AuthorityKeyIdentifierExtension(
+            byte[] subjectKeyIdentifier
+            )
+        {
+            if (subjectKeyIdentifier == null) throw new ArgumentNullException(nameof(subjectKeyIdentifier));
+            m_keyIdentifier = subjectKeyIdentifier;
+            base.Oid = new Oid(AuthorityKeyIdentifier2Oid, kFriendlyName);
+            base.Critical = false;
+            base.RawData = Encode();
+        }
+
+        /// <summary>
+        /// Build the X509 Authority Key extension.
+        /// </summary>
+        /// <param name="subjectKeyIdentifier">The subject key identifier</param>
+        /// <param name="authorityName">The distinguished name of the issuer</param>
+        /// <param name="serialNumber">The serial number of the issuer certificate</param>
+        public X509AuthorityKeyIdentifierExtension(
+            byte[] subjectKeyIdentifier,
+            X500DistinguishedName authorityName,
+            byte[] serialNumber
+            )
+        {
+            if (subjectKeyIdentifier == null) throw new ArgumentNullException(nameof(subjectKeyIdentifier));
+            if (authorityName == null) throw new ArgumentNullException(nameof(authorityName));
+            if (serialNumber == null) throw new ArgumentNullException(nameof(serialNumber));
+            m_issuer = authorityName;
+            m_keyIdentifier = subjectKeyIdentifier;
+            m_serialNumber = serialNumber;
+            base.Oid = new Oid(AuthorityKeyIdentifier2Oid, kFriendlyName);
+            base.Critical = false;
+            base.RawData = Encode();
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension(Oid oid, byte[] rawData, bool critical)
+        :
+            base(oid, rawData, critical)
+        {
+            Decode(rawData);
+        }
+        #endregion
+
+        #region Overridden Methods
+        /// <summary>
+        /// Returns a formatted version of the Authority Key Identifier as a string.
+        /// </summary>
+        public override string Format(bool multiLine)
+        {
+            StringBuilder buffer = new StringBuilder();
+
+            if (m_keyIdentifier != null && m_keyIdentifier.Length > 0)
+            {
+                if (buffer.Length > 0)
+                {
+                    if (multiLine)
+                    {
+                        buffer.Append("\r\n");
```

Comment:
Just clarifying that you're intentionally always using Windows newlines instead of the OS-relevant newlines that buffer.AppendLine() would do.

(The existing X509Extension classes use AppendLine, or native libraries that use their OS newline)

Reply:
all appendline fixed